lib/wrapeth/provider/interface.ex

```diff
@@ -0,0 +1,73 @@
+defmodule Wrapeth.Provider.Interface do
+  alias Wrapeth.Provider.Behaviour
+
+  defmacro __using__(opts) do
+    quote bind_quoted: [opts: opts] do
+      @behaviour Behaviour
+
+      @otp_app opts[:otp_app]
+      @app_name opts[:app_name]
+
+      def get_module_and_url() do
+        config = Application.get_env(@app_name, @otp_app)
+
+        case config[:client_type] do
+          :http -> {Ethereumex.HttpClient, config[:node_url]}
+          :ipc -> {Ethereumex.IpcClient, config[:ipc_path]}
+          _ -> {:error, :invalid_client_type}
+        end
+      end
+
+      def get_accounts() do
```

Comment:
If this is an implementation of the `Behaviour`, we should be explicit about it.

Reply:
I think that it is solved in the new commit.😁